### ISSUE TITLE
`if self.meta` is no longer equivalent to checking an object has defined raster metadata

### ISSUE THREAD

**Issue by ChrisFairless** [CLIMADA-project/climada_python#340]
Historically the `meta` property of CLIMADA classes was used only to store raster metadata. So if you wanted to check that an object had a defined raster transformation you could just check for the existence of its `meta` property.

But there's been a bit of scope creep, and now `Exposures.meta` often stores CRS information for Exposures without raster metadata, and after appending or union-ing Centroids and Hazards their `meta` is set to an empty dictionary.

So this means that in certain situations the (very common) check
```
if x.meta:
    ...
```
no longer confirms the existence of raster metadata. See for example PR #337 which is how I started thinking about this.

Basically we need to read through the codebase (especially Centroids, Hazard and Exposures) and look for this check failing (but I don't have time right now.)

We could solve it by writing a `has_raster_metadata` method to replace the check, or by enforcing stricter data structures on objects' `meta` properties.

**Comment by chahank** [comment]
I agree that this might be a problem. However, I am not sure that a quick fix like `has_raster_metadata` is the optimal path. The handling of `meta` in CLIMADA in general is at points inconsistent, and maybe it would be worth the effort to find a more encompassing solution. 

Could you maybe start by making a list of all the methods and classes that are using the `meta` attribute? Then we might have an overall view of the problem.

**Comment by ChrisFairless** [comment]
Closing this and moving the discussion to issue #348
